modules/phase_field/src/kernels/CHSplitFlux.C

```diff
@@ -0,0 +1,55 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "CHSplitFlux.h"
+
+template<>
+InputParameters validParams<CHSplitFlux>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Computes flux as nodal variable - flux = -mobility * grad(chemical_potenstial)");
```

Comment:
Creep strain increment is a material property that depends on the gradient of flux. By making flux as a nodal variable, the order of element shape functions can be reduced. Also for a spatially varying mobility/diffusivity, the chain-rule on derivatives can be avoided.